nconnect.go

```diff
@@ -311,29 +316,38 @@ func (nc *nconnect) StartClient() error {
 
 	remoteTunnelAddr := nc.opts.RemoteTunnelAddr
 	if len(remoteTunnelAddr) == 0 {
-		remoteInfo, err := nc.getRemoteInfo()
-		if err != nil {
-			return err
+		for _, remoteAdminAddr := range nc.opts.RemoteAdminAddr {
+			remoteInfo, err := nc.getRemoteInfo(remoteAdminAddr)
+			if err != nil {
+				log.Printf("getRemoteInfo %v err: %v\n", remoteAdminAddr, err)
+				continue
+			}
+			remoteTunnelAddr = append(remoteTunnelAddr, remoteInfo.Addr)
 		}
-		remoteTunnelAddr = remoteInfo.Addr
+	}
+	if len(remoteTunnelAddr) == 0 {
+		return fmt.Errorf("no remote tunnel address, start client fail")
 	}
 
 	var vpnCIDR []*net.IPNet
 	if nc.opts.VPN {
 		vpnRoutes := nc.opts.VPNRoute
 		if len(vpnRoutes) == 0 {
-			remoteInfo, err := nc.getRemoteInfo()
-			if err != nil {
-				return err
-			}
-			if len(remoteInfo.LocalIP.Ipv4) > 0 {
-				vpnRoutes = make([]string, 0, len(remoteInfo.LocalIP.Ipv4))
-				for _, ip := range remoteInfo.LocalIP.Ipv4 {
-					if ip == nc.opts.TunAddr || ip == nc.opts.TunGateway {
-						log.Printf("Skipping server's local IP %s in routes", ip)
-						continue
+			for _, remoteAdminAddr := range nc.opts.RemoteAdminAddr {
+				remoteInfo, err := nc.getRemoteInfo(remoteAdminAddr)
+				if err != nil {
+					return err
```

Comment:
The logic here is inconsistent with L323.
In L323 (get remote info), failure is printed and remote addr skipped.
Here any failure will cause the app to stop.

Reply:
Yes, modified.